### PR TITLE
refactor and fix github provider

### PR DESCRIPTION
**Pull request by chihangwang**

### CODE REVIEW

File: src/github/provider.py [hunk 1]
```diff
@@ -7,11 +8,11 @@
 from src.github.models import CommentType, PRComment, PullRequest, FileDiff
 
 class GithubProvider:
-    def __init__(self, owner: str, repo: str | None = None, token: str | None = None):
```
Comment (chihangwang) [inline src/github/provider.py:10]: We only need `owner` and `token` to initialize the provider. As for `repo`, it's left to be decided in the methods later.

File: src/github/provider.py [hunk 2]
```diff
@@ -144,22 +145,8 @@ def fetch_pr(self, pr_number: int) -> PullRequest:
         Returns:
             PullRequest: The `PullRequest` object containing detailed information about the PR.
         """
-        pr = self.api.pulls.get(pr_number)
+        pr = self.api.pulls.get(repo, pr_number)
         return self._to_PullRequest(pr)
-    
-    def list_prs(self, per_page: int, n_pages: int) -> list[PullRequest]:
```
Comment (chihangwang) [inline src/github/provider.py:150]: This method isn't used anywhere. Also, it's not very useful unless it has `repo` and some filters.

File: src/github/provider.py [hunk 3]
```diff
@@ -78,33 +79,33 @@ def get_patch(self, pr_file: dict) -> str:
...
-            repo_name (str): The name of the repository.
             pr (dict): The pull request data from the GitHub API.
-            username (str): The username of the author to match.
 
         Returns:
             PullRequest: The `PullRequest` object containing detailed information about the PR.
         """
         pr_number = pr['number']
-        pr_files = self.api.pulls.list_files(owner=owner, repo=repo_name, pull_number=pr_number)
+        owner = pr['user']['login']
```
Comment (henry60603) [inline src/github/provider.py]: is `owner` used in the following code? or do we use the one in the self?

File: src/github/provider.py (hunk 2 above)
Comment (henry60603) [reply src/github/provider.py:150]: it is used in the previous pipeline. 
let's remove it!

File: src/github/provider.py [hunk 4]
```diff
@@ -237,3 +224,8 @@ def download_user_repo_prs(self, owner: str, usernames: list[str], start_date: d
                     with open(file_path, "w") as save_file:
                         json.dump(pullRequests_dict, save_file, indent=6)
                     print(f'Saved PRs for {username} in {file_name}')                      
+
+if __name__ == '__main__':
```
Comment (henry60603) [inline src/github/provider.py]: what's this used for?

File: src/github/provider.py (hunk 3 above)
Comment (chihangwang) [reply src/github/provider.py]: Nice catch

File: src/github/provider.py (hunk 4 above)
Comment (chihangwang) [reply src/github/provider.py]: It's just testing to make sure things work. I'd comment them out.

File: src/github/provider.py (hunk 2 above)
Comment (chihangwang) [reply src/github/provider.py:150]: Where is it still used? I couldn't find it.

Comment (henry60603) [reply src/github/provider.py:150]: Oh I mean it is not in used now.
We can remove it!

**Review by henry60603 — APPROVED** [review]
LGTM

**Pull request merged**